### DISCUSSION TITLE
Hardware recommendation

### DISCUSSION THREAD

**Discussion by pluberd** [Comfy-Org/ComfyUI#5500] [General]
I'm planning to upgrade my hardware, but I'm finding that it's not easy to make an informed decision.

I currently have an upgraded Dell Presicion Tower with Inter i7 6700 CPU, 4*16 GB DDR 4 2133 RAM, RTX 3090 and a 750W power supply.

I use the computer as a headless server mainly to run open source models locally. Comfyui is one of my main applications. It happens that one or two VMs are running in parallel.

 As should be known, this increasingly causes models to change and need to be loaded quickly. Also it seems to me that in some tasks, the CPU creates load, which seems to form a bottleneck.

I initially plan to continue using my RTX 3090, but I would definitely like to be able to upgrade later and continue to use the 3090 in parallel if that makes sense.

I think that I need to replace the following components:

- CPU / motherboard / RAM
- Case, power supply

I would like to continue using the m.2 SSD that I already have, even if I buy a new one.

My budget is 2-3K EUR. I don't need any lighting effects, I'll never pose with the computer or use computer games.

A tolerable volume and moderate power consumption are important to me. I don't train any models, so I probably only generate load peaks. If necessary, I would like to have the power I need.

But I don't want to spend money to be at the top of my game. A price/performance ratio is important to me.

I think the first thing would be to decide on a CPU and then get a suitable motherboard and memory.

But what would make sense for me? I've had my eye on the Intel i7-12700 for a while. But I'm totally unsure.

**Comment by pluberd** [comment]
After doing some research, I'm leaning towards the Ryzen 7 9700X with an X670E motherboard and 2x48GB DDR5-5600 RAM. That seems to me to give a good price-performance ratio.

8 CPU cores are a good compromise and I won't need the X3D cache in my use case.

Including the case, I etc. bin at about 1200€ and still have room for a really fast SSD.

**Comment by keremcozin** [comment]
I'm using ComfyUI and checkpoints/models with NVIDIA RTX 2050 notebook, it has 4gb vram, it's not much for these tasks but I can assure it works great for them. A simple AI generation for 512x768 finishes in 2-3 minutes at most with good quality output.

**Comment by Rogala** [comment]
Motherboard on the z-790 chipset, supports PCI-E 5.0 x16, 4 NVMe slots and more PCIe 4.0 lanes than on the z-690 (20 vs 12), CPU 7-13700 supports faster 5600 memory, not 4800 on 7-12700, RAM 4x32GB 5600 and 2x2TB NVMe 4.0 x4, will cost about 1200-1400 euros.

**Comment by pluberd** [comment]
Because of the problem with stability, I decided against a CPU from Intel in generation 13 or 14. The Ryzen 9700x I chose is comparable in price and performance, but it only requires half the electrical power.

Furthermore, I decided on the board "MSI Tomahawk WIFI AMD X670E". At first I was skeptical if it was a good choice, but eventually it goes surprisingly well. My monitors are connected to the internal GPU and my RTX is therefore fully available for AI applications or can even be passed through to a VM.

At the moment I am very satisfied.

**Comment by uberneko** [comment]
If I can give one recommendation, it is to have a CPU with integrated graphics.
I still don't know if it is normal or not, but AI puts graphic cards in a quite high stress, and I have had some freezes if the GPU is also in charge of the OS desktop..

(soon or later 9700X will be my choice as well)
